### DISCUSSION TITLE
Is swc faster then bun?

### DISCUSSION THREAD

**Discussion by Milo123459** [swc-project/swc#2743] [General]
Bun: https://bun.sh

I tried it out and it is very, very fast. It doesn't seem particularly mature yet, and was curious if such speeds of bun could be adopted into SWC, to be very fast, and be as mature as this project is.

It is written in Zig, which is fast (but because Rust is yet again, more mature) but can be overtaken.

It's closed source right now, but to get access to it you just have to join the Discord listed on their website and type `i want bun` in the Discord and it'll invite you to the repo.

Thanks! 👋

**Comment by kdy1** [comment]
I'm not sure, but I prefer to focus on making `swc` extensible/reusable. If there are some good tricks that do not affect users badly, I'll implement them in swc.

Being fast is important, but it was not the highest priority and it'll never be. 

If you don't care about reusability, you can use lots of tricks in lexer and it will be super performant. But I don't use such tricks and instead, I always try to make modules generally usable.

**Reply by Milo123459** [reply]
I completely understand, and I think that is completely reasonable. I have been working on something (for like a week or so) that just removes some un-needed allocations to make everything a bit faster, and that PR should hopefully be done soon;
